csharp/src/Ice/Reference.cs

```diff
@@ -1344,13 +1344,13 @@ internal async ValueTask<IRequestHandler> GetConnectionRequestHandlerAsync()
                         Communicator.Logger.Trace(traceLevels.RetryCat, "connection to cached endpoints failed\n" +
                             $"removing endpoints from cache and trying again\n{ex}");
                     }
-                    return await GetConnectionRequestHandlerAsync();
+                    return await GetConnectionRequestHandlerAsync(cancel);
                 }
                 throw;
             }
         }
 
-        internal async ValueTask<IRequestHandler> GetRequestHandlerAsync()
+        internal async ValueTask<IRequestHandler> GetRequestHandlerAsync(CancellationToken cancel = default)
```

Comment:
`= default` makes it more likely we accidentally forget this argument.

Reply:
Right, I'm not sure where to draw the line. I guess we could make it mandatory for all our internal APIs where cancellation needs to be handled. Here it's optional because the synchronous GetConnection method doesn't need it but specifying it explicitly would indeed probably be better, I'll fix it.